lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -426,7 +426,7 @@ JavaScriptCompiler.prototype = {
     if (this.isInline()) {
       this.replaceStack(current => [' != null ? ', current, ' : ""']);
 
-      this.pushSource(this.appendToBuffer(this.popStack()));
+      this.pushSource(this.appendToBuffer([this.popStack(), '+', "''"]));
```

Comment:
This changes the output of compiled templates, so tests have to be updated. Or a solution is found without altering templates.

Reply:
I don't see a way to do this without template changes. But we also don't need this in all places where "append" is called.
I believe BlockStatements and  PartialStatements are not affected because they always return strings.

So your solution looks simple, but I could probably combine `'+', "''"` into `"+''"`

Are the other tests passing? Seems like that they are not even run.

Reply:
Thanks for the input!

There is the `test:bin` tests that check if the compiled templates match the expected compilation-output. I have to adapt those tests.

Reply:
I fixed the `test:bin` tests.